### PR TITLE
Add siren neo nas-ab02b2

### PR DESCRIPTION
**Pull request by William-De71**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [ ] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([french forum](https://community.gladysassistant.com/)/[english forum](https://en-community.gladysassistant.com/)) for testing before merging.
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Add all expose of Neo siren model NAS AB02B2

image

image

### CODE REVIEW

**Comment by Pierre-Gilles** [comment]
@atrovato I'm curious to have your feedback on this PR :) Especially on the way that the "melody" enum is handled

File: front/src/components/boxs/device-in-room/device-features/VolumeDeviceFeature.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import get from 'get-value';
+import { Text } from 'preact-i18n';
+import cx from 'classnames';
+
+import { DeviceFeatureCategoriesIcon } from '../../../../utils/consts';
+import { SIREN_VOLUME } from '../../../../../../server/utils/constants';
+
+const VolumeDeviceFeature = ({ children, ...props }) => {
```
Comment (atrovato) [inline front/src/components/boxs/device-in-room/device-features/VolumeDeviceFeature.jsx]: I'm not really sure that "low/medium/high" will be the only volume management for sirens.
I think we can have such 0 to 100% volume level.
Maybe we should setup a specific "LMH_VOLUME" feature?

File: server/services/zigbee2mqtt/exposes/enumType.js [hunk 2]
```diff
@@ -111,6 +125,20 @@ module.exports = {
...
+        category: DEVICE_FEATURE_CATEGORIES.SIREN,
+        type: DEVICE_FEATURE_TYPES.SIREN.VOLUME,
+      },
+    },
+    melody: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.SIREN,
+        type: DEVICE_FEATURE_TYPES.SIREN.MELODY,
+        min: 1,
+        max: 18,
```
Comment (atrovato) [inline server/services/zigbee2mqtt/exposes/enumType.js]: Why did you set min/max values? If this attributes are provided by z2m device, remove them from here.

**Review by atrovato — COMMENTED** [review]
I have some comments:
 1. are we sure to manage only the low/medium/high siren volume? I think some device (out of z2m) will be a 0-100% volume management. Maybe we should rename the feature to be more explicit on "low/medium/high" values ; "volume" is maybe too generic.
 2. melody mgmt looks great according to z2m specifications, but, according to my feeling, it will introduce the "enum values" on gladys features to be able to handle named values (but keep working with numerics) cc @Pierre-Gilles

File: front/src/components/boxs/device-in-room/device-features/VolumeDeviceFeature.jsx (hunk 1 above)
Comment (William-De71) [reply front/src/components/boxs/device-in-room/device-features/VolumeDeviceFeature.jsx]: I'll see to create a new volume feature as your suggestion

File: server/services/zigbee2mqtt/exposes/enumType.js (hunk 2 above)
Comment (William-De71) [reply server/services/zigbee2mqtt/exposes/enumType.js]: OK I'll remove it

**Comment by William-De71** [comment]
@atrovato  

> melody mgmt looks great according to z2m specifications, but, according to my feeling, it will introduce the "enum values" on gladys features to be able to handle named values (but keep working with numerics) 

And how do you think we handle that?

**Comment by Pierre-Gilles** [comment]
@William-De71 Let me know (on the forum) when @atrovato feedbacks are fixed :) 

> it will introduce the "enum values" on gladys features 

For that, it's not really the topic here imo. If the melody management is good for you @atrovato, I'm good to merge after the 2 feedbacks are corrected

File: front/src/components/boxs/device-in-room/SupportedFeatureTypes.jsx [hunk 3]
```diff
@@ -12,7 +12,10 @@ const SUPPORTED_FEATURE_TYPES = [
   DEVICE_FEATURE_TYPES.SHUTTER.STATE,
   DEVICE_FEATURE_TYPES.THERMOSTAT.TARGET_TEMPERATURE,
   DEVICE_FEATURE_TYPES.AIR_CONDITIONING.MODE,
-  DEVICE_FEATURE_TYPES.AIR_CONDITIONING.TARGET_TEMPERATURE
+  DEVICE_FEATURE_TYPES.AIR_CONDITIONING.TARGET_TEMPERATURE,
+  DEVICE_FEATURE_TYPES.SIREN.VOLUME,
```
Comment (Pierre-Gilles) [inline front/src/components/boxs/device-in-room/SupportedFeatureTypes.jsx]: LMH_VOLUME no?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]

File: front/src/components/boxs/device-in-room/SupportedFeatureTypes.jsx (hunk 3 above)
Comment (William-De71) [reply front/src/components/boxs/device-in-room/SupportedFeatureTypes.jsx]: you are right. Fixed!

**Review by Pierre-Gilles — APPROVED** [review]

**Pull request merged**